ast/compile_test.go

```diff
@@ -4232,3 +4232,23 @@ deny {
 		t.Fatalf("Expected error for unsafe built-in but got %v", err)
 	}
 }
+
+func TestCompilerPassesTypeCheck(t *testing.T) {
+	c := NewCompiler().
+		WithCapabilities(&Capabilities{Builtins: []*Builtin{Split}})
+	// Must compile to initialize type environment after WithCapabilities
+	c.Compile(nil)
```

Comment:
💭 It's little awkward to have to have to call this, and spread the comment -- I guess we could change the internals of compiler, too. Anyways, since the non-test usecase seems to not require that, it's OK.